lib/commands/touch.js

```diff
@@ -1,9 +1,51 @@
 const commands = {};
-
 //This is needed to make clicks on -image elements work properly
 commands.performActions = async function (actions) {
-  return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  switch (actions[0].id) {
+    case 'default keyboard':
+      this.log.info('Mapping keyboard actions');
+      return await performKeyboardCommands(this.winAppDriver, actions[0].actions);
+    case 'default mouse':
+      this.log.info('Mapping mouse actions');
+      return await performMouseCommands(this.winAppDriver, actions[0].actions, this.log);
+    default:
+      this.log.info('Going default actions route');
+      return await this.winAppDriver.sendCommand('/actions', 'POST', {actions});
+  };
 };
 
+async function performKeyboardCommands(winAppDriver, actions) {
+  const keysFormatted = formatKeyboardInput(actions);
+  return await winAppDriver.sendCommand('/keys', 'POST', {'value': keysFormatted });
+}
+
+async function performMouseCommands(winAppDriver, actions, log) {
+  let result;
+  for (let action in actions) {
+    switch (actions[action].type) {
+      case 'pointerMove':
```

Comment:
pauses should not be ignored

Reply:
pauses aren't under a command with mouse input device key, so they will go through the default /actions route

Reply:
Had another look, pause is coming an an action but there is no winAppDriver endpoint I can proxy it to...:(

Reply:
It looks like you need to pause in the code before sending the next action, similar to https://github.com/microsoft/WinAppDriver/blob/master/Tests/WebDriverAPI/TouchDownMoveUp.cs does

Reply:
Done